CMakeLists.txt

```
@@ -2,59 +2,54 @@
 # pynest2d is released under the terms of the LGPLv3 or higher.
 
 project(pynest2d)
-cmake_minimum_required(VERSION 3.6)  # Lowest version it's been tested with.
+cmake_minimum_required(VERSION 3.18)  # Lowest version it's been tested with.
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
-# Requirements.
-# FIXME: Remove the code for CMake <3.12 once we have switched over completely.
-# FindPython3 is a new module since CMake 3.12. It deprecates FindPythonInterp and FindPythonLibs.
-if(${CMAKE_VERSION} VERSION_LESS 3.12)
-    # FIXME: Use FindPython3 to find Python, new in CMake 3.12.
-    # However currently on our CI server it finds the wrong Python version and then doesn't find the headers.
-    find_package(PythonInterp 3.5 REQUIRED)
-    find_package(PythonLibs 3.5 REQUIRED)
-
-else()
-    # Use FindPython3 for CMake >=3.12
-    find_package(Python3 3.5 REQUIRED COMPONENTS Interpreter Development)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+if(NOT DEFINED Python_VERSION)
+    set(Python_VERSION
+            3.10
+            CACHE STRING "Python Version" FORCE)
+    message(STATUS "Setting Python version to ${Python_VERSION}. Set Python_VERSION if you want to compile against an other version.")
+endif()
+if(APPLE)
+    set(Python_FIND_FRAMEWORK NEVER)
```

```suggestion
    set(Python_FIND_FRAMEWORK NEVER)
```

ALready mentioned this for the libSavitar changes. We seem to have a good reason to do this, but it could also well be that we're just copying it from other files since "that how we have been doing it in the past". A bit of documentation to clarify why we do this would be good.